skyportal/tests/frontend/test_plotly_plots_frontend.py

```diff
@@ -0,0 +1,37 @@
+import time
+import pytest
+from selenium.webdriver.common.by import By
+
+from baselayer.app.config import load_config
+
+cfg = load_config()
+
+
+def expand_shadow_element(driver, element):
+    shadow_root = driver.execute_script('return arguments[0].shadowRoot', element)
+    return shadow_root
+
+
+@pytest.mark.flaky(reruns=3)
+def test_export_bold_light_curve_as_csv_button(driver, user, public_source):
+    driver.get(f"/become_user/{user.id}")
+    driver.get(f"/source/{public_source.id}")
+
+    button_present = False
+    nretries = 0
+    while button_present is False and nretries < 20:
```

Comment:
Do you have any suggestions? The idea is that our page has N number of elements that render in different order and speeds based on the availability of the API, the browser used, ... a ton of different factors.

Ideally, we should open a PR and replace all of the things that need to load by skeletons that ensure a fixed size before we load the data. That will allow us not to have these retries to check that something loads.

Because if we trade this pattern for an explicit wait that everything else has loaded, it might be a little tricky as it depends for each test... This is why I had a time.sleep() above, which I traded for a while loop so that if it works on the first try we don't wait for nothing.